joplin/static/js/admin.js

```diff
@@ -81,45 +81,88 @@ function configurePreviewUpdates() {
     return;
   }
 
-  editForm.addEventListener('input', _.debounce(updatePreviewWindow, 800));
+  const updateFn = _.debounce(updatePreviewViaSession, 800)
+  editForm.addEventListener('input', updateFn);
+  // This ensures we get updates for added snippets and other things the wagtail admin does that hides normal form flow
+  editForm.addEventListener('DOMSubtreeModified', updateFn);
+  updateFn();
 }
 
-function updatePreviewWindow() {
-  let controls = document.querySelector('form#page-edit-form').elements;
-  let d = {};
-  for(let i = 0; i < controls.length; i++) {
-    let el = controls[i];
-    let elType = el.type;
-    if (!el.name || el.hidden || elType === 'submit' || el.name.endsWith('-count')) {
-      continue;
-    }
+function updatePreviewViaSession() {
+  const form = document.querySelector('#page-edit-form');
+  const button = form.querySelector('.action-preview');
+  const previewURL = button.dataset.action;
 
-    let path = el.name.split('-').reduce((accumulatedVal, key) => {
-      return accumulatedVal + (Number.isNaN(Number.parseInt(key)) ? `.${key}` : `[${key}]`)
-    });
-    console.debug(`${elType} ${el.name} = ${el.value} (path = ${path})`);
+  const baseOptions = {
+    method: 'POST',
+    credentials: 'same-origin',
+  }
 
-    if (elType === 'select') {
-      _.set(d, path, {
-        id: el.value,
-        text: encodeURIComponent(el.options[el.selectedIndex].text),
-      })
+  const pageID = window.location.href.match(/pages\/(\d+)\/edit/).pop();
+  const graphqlBody = `{
```

Comment:
🔥

Reply:
#blessed are those who get to read this js later